StyleCop.Analyzers/StyleCop.Analyzers/Helpers/AccessLevelHelper.cs

```diff
@@ -78,5 +81,508 @@ internal static string GetName(AccessLevel accessLevel)
+            {
+            case AccessLevel.Public:
+                return Accessibility.Public;
+
+            case AccessLevel.Internal:
+                return Accessibility.Internal;
+
+            case AccessLevel.ProtectedInternal:
+                return Accessibility.ProtectedOrInternal;
+
+            case AccessLevel.Protected:
+                return Accessibility.Protected;
+
+            case AccessLevel.Private:
+                return Accessibility.Private;
+
+            case AccessLevel.NotSpecified:
+            default:
+                break;
+            }
+
+            if (!syntax.Modifiers.Any(SyntaxKind.PartialKeyword))
+            {
+                BaseTypeDeclarationSyntax enclosingType = syntax.Parent as BaseTypeDeclarationSyntax;
+                return enclosingType == null ? Accessibility.Internal : Accessibility.Private;
+            }
+
+            INamedTypeSymbol declaredSymbol = semanticModel.GetDeclaredSymbol(syntax, cancellationToken);
+            return declaredSymbol?.DeclaredAccessibility ?? Accessibility.NotApplicable;
+        }
+
+        internal static Accessibility GetDeclaredAccessibility(this BaseMethodDeclarationSyntax syntax, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (syntax == null)
+            {
+                throw new ArgumentNullException(nameof(syntax));
+            }
+
+            if (semanticModel == null)
+            {
+                throw new ArgumentNullException(nameof(semanticModel));
+            }
+
+            AccessLevel accessLevel = GetAccessLevel(syntax.Modifiers);
+            switch (accessLevel)
+            {
+            case AccessLevel.Public:
+                return Accessibility.Public;
+
+            case AccessLevel.Internal:
+                return Accessibility.Internal;
+
+            case AccessLevel.ProtectedInternal:
+                return Accessibility.ProtectedOrInternal;
+
+            case AccessLevel.Protected:
+                return Accessibility.Protected;
+
+            case AccessLevel.Private:
+                return Accessibility.Private;
+
+            case AccessLevel.NotSpecified:
+            default:
+                break;
+            }
+
+            if (syntax.Modifiers.Any(SyntaxKind.PartialKeyword))
+            {
+                return Accessibility.Private;
+            }
+
+            MethodDeclarationSyntax methodDeclarationSyntax = syntax as MethodDeclarationSyntax;
+            if (methodDeclarationSyntax != null)
+            {
+                if (methodDeclarationSyntax.ExplicitInterfaceSpecifier == null)
+                {
+                    return Accessibility.Private;
+                }
+                else
+                {
+                    return Accessibility.Public;
+                }
+            }
+
+            if (syntax.IsKind(SyntaxKind.ConstructorDeclaration))
+            {
+                return Accessibility.Private;
```

Comment:
Since this rule isn't used for ordering, we can be more strict here. I think treating them as private is a more accurate filter on which teams would require they be documented, especially considering the documentation of static constructors almost never gets included in the published output.